### ISSUE TITLE
$0 trial needs full CC verification to avoid fraud abuse

### ISSUE THREAD

**Issue by lkraav** [woocommerce/woocommerce-gateway-stripe#1155]
**Is your feature request related to a problem? Please describe.**
We have $0 7-day subscription trial configured via WooCommerce Subscriptions, but it's getting abused by scam accounts and fraudulent credit cards, because no verification happens on initial signup.

It's too late when the first real charge attempt occurs.

**Describe the solution you'd like**
We need full credit card verification checks to happen on $0 trial signup.

**Describe alternatives you've considered**
Force $1 trial instead of $0, but it potentially creates more friction than we'd like.

**Comment by thenbrent** [comment]
@lkraav thanks for raising this. Can you please confirm, you don't have the [**Allow $0 initial checkout without a payment method** setting](https://docs.woocommerce.com/document/subscriptions/version-2-5/#section-2) enabled, and subscribers are putting in credit cards at the time of checkout?

(And the issue is that fraudsters are using stolen cards or some other type of credit card which fails to process a payment.)

**Comment by lkraav** [comment]
Hi @thenbrent. Yes, we require payment methods on all trials. They are inputting credit cards, we see the Sources recorded in Stripe. Another confirmational indicator is that without payment methods on record, the first full charge "renewal" wouldn't even produce any credit card error codes.

> And the issue is that fraudsters are using stolen cards or some other type of credit card which fails to process a payment.

Yes, that's what it boils down to. We saw many "expired_card" errors for one, a type of error that definitely should never pass initial trial setup.

We're currently working around this by going with $1 trial instead. It increases friction on one hand, but maybe also bumps lead quality somewhat due to basic penny gap psychology.

Regardless, having an unscammable (or less-scammable) $0 trial setup in the product team marketing arsenal would also be helpful.

Stripe support wrote that extra verification checks are possible via API, even for $0 transactions. WooCommerce support claimed that skipping verification for $0 transactions is part of card network speed optimization strategy. Not sure who to believe, yet, hence this issue.

**Comment by vbelolapotkov** [comment]
Marking the priority for this issue as low for now because not that many users are reported a similar problem. Added it to one of the next milestones (4.4.2) for consideration.

**Comment by v18** [comment]
I was surprised to see this issue open up because we added SetupIntents support last year, so I took a look. The current behaviour is that the Stripe gateway _does_ ask for 3DS authentication before allowing the subscription to be purchased.

1. Set up a new subscription with a free trial
2. Add that subscription to cart
3. Check out with a 3DS card that will fail all transactions, such as `4000008260003178` ([doc with more test cards](https://stripe.com/docs/testing#regulatory-cards))
4. See that you are asked to authenticate the transaction
5. If you fail to authenticate the payment, see that the purchase fails and an error message is displayed

Similar steps can be taken to see that the purchase fails with a test card where all payments are declined, like `4000000000000002`.

I'm inclined to close this issue for now because it's possible to reproduce. We can re-open if we can find steps that show an incorrect behaviour happening.

**Comment by v18** [comment]
@lkraav, I wanted to check with you specifically about the problem that you're having. Is the issue that you're getting too many subscribers, who use the free trial and do not upgrade later? Is the problem that too many of the subscribers are charged automatically at the due date of the subscription renewal, but the charge on the credit card fails?

**Comment by DanReyLop** [comment]
> I also tried with the card 4000000000000069, which returns the error expired_card. Following the steps above with this card also results in a failed checkout.

I think that a better test would be to use these cards:

Card | Description
-- | --
`4000000000009995` | Charge is declined with a `card_declined` code. The decline_code attribute is `insufficient_funds`.
`4000000000009987` | Charge is declined with a `card_declined` code. The decline_code attribute is `lost_card`.
`4000000000009979` | Charge is declined with a `card_declined` code. The decline_code attribute is stolen_card.

I tried all of them and the checkout fails, even with a $0 charge. So there **is** validation being performed. Maybe real-life credit cards don't fail as reliably on $0 charges? I had asked Stripe a while ago about this, I'll follow up.

**Comment by lkraav** [comment]
> I wanted to check with you specifically about the problem that you're having. Is the issue that you're getting too many subscribers, who use the free trial and do not upgrade later?

No. :arrow_up: would be a business problem, not a technical one I opened the issue for.

> Is the problem that too many of the subscribers are charged automatically at the due date of the subscription renewal, but the charge on the credit card fails?

"Too many of the subscribers" at general customer base level is not the issue. This issue is filed specifically to address fraud intent segment(s) :arrow_down: 

> The current behaviour is that the Stripe gateway does ask for 3DS authentication before allowing the subscription to be purchased.

Fraudsters use legit, active credit cards, where

* I'm not sure if advanced protection mechanisms like 3DS trigger at all, cards themselves may not have it enabled. Stripe docs say "Not all cards support 3D Secure or require the customer be redirected to their card issuer's authentication page."

* I'm not sure if billing address, CVC etc is verified at $0 transaction

* As soon as we switched to $1 trial, forcing all CC-s go through an actual charge, incident count is near-zero if not zero. There's a naked eye visible difference.
